### PR TITLE
Python style: update yapf to 0.30.0, reformat code

### PR DESCRIPTION
**Pull request by sergiitk**
yapf 0.30.0 fixed an issue with over-indenting a parameter list when not needed. This issue was breaking [Google Python style](https://google.github.io/styleguide/pyguide.html#3192-line-breaking), and decreased code readability.

This PR updates yapf from `0.28.0` to `0.30.0` ([CHANGELOG](https://github.com/google/yapf/blob/master/CHANGELOG)) and re-formats the code using updated yapf.

### CODE REVIEW

**Comment by sergiitk** [comment]
Let me know if a PR like this makes sense. If yes, I'll proceed with fixing "Sanity Checks (internal CI)".

**Comment by lidizheng** [comment]
PR like this looks fine to me, including the change and reformat. Taming both pylint and yapf can be tricky.

**Comment by sergiitk** [comment]
Thanks! I'll give fixing pylint a try. Really looking forward to this change.

**Comment by sergiitk** [comment]
Force push note:
- split upgrade to 0.30 and code reformat to individual commits
- rebased against the latest master

**Comment by lidizheng** [comment]
@sergiitk A gentle reminder in case you missed the CI result. Pylint is still conflicting with the format of newer YAPF. Generally, YAPF is not the best maintained formatter (see https://news.ycombinator.com/item?id=17155048):

> YAPF would at times not produce deterministic formatting (formatting the same file the second time with no changes in between would create a different formatting)

As for the next step, I would recommend that we completely suppress the pylint complain about indentation, after all, pylint is better at analysis bugs and code quality then code format. CC @gnossen

**Comment by sergiitk** [comment]
Hey @lidizheng, I've done a bunch of experimenting that day with updating pylint, since pylint improved its behavior with `bad-continuation` and in the latest version didn't produce this error anymore. However, it's started producing whole bunch more. So I didn't get a quick win and had to put this on a backburner to finish another project. It's done so today have time to follow up on this.
I'll do another rebase and suppress `bad-continuation` for now. We should update `pylint` at some point, but as a separate commitment.

**Comment by sergiitk** [comment]
Alright! Sanity check pass now.

@lidizheng this is ready for a review now. Note: commit 88af427 is exactly the situation with yapf instability you described earlier. I've had to disambiguate it manually. The error is seen in [this build](https://source.cloud.google.com/results/invocations/aaa1c0ad-ff69-48a7-9b0a-408c19e8a6f2/targets/github%2Fgrpc%2Frun_tests%2Fsanity_linux_dbg_native%2Ftools%2Fdistrib%2Fyapf_code.sh/tests).

File: tools/distrib/yapf_code.sh [hunk 1]
```diff
@@ -33,8 +33,6 @@ VIRTUALENV=yapf_virtual_environment
 
 python3 -m virtualenv $VIRTUALENV -p $(which python3)
 PYTHON=${VIRTUALENV}/bin/python
-"$PYTHON" -m pip install --upgrade pip==19.3.1
-"$PYTHON" -m pip install --upgrade futures
-"$PYTHON" -m pip install yapf==0.28.0
+"$PYTHON" -m pip install yapf==0.30.0
 
-$PYTHON -m yapf $ACTION --recursive --style=setup.cfg "${DIRS[@]}"
+$PYTHON -m yapf $ACTION --parallel --recursive --style=setup.cfg "${DIRS[@]}"
```
Comment (lidizheng) [inline tools/distrib/yapf_code.sh:38]: Nice addition!

**Review by lidizheng — APPROVED** [review]
LGTM

**Comment by sergiitk** [comment]
Another note on 88af427: I believe this change, generator to list comprehension, is a no-op, since the generator is immediately unpacked with the `*`-operator anyway.

File: tools/distrib/yapf_code.sh (hunk 1 above)
Comment (sergiitk) [reply tools/distrib/yapf_code.sh:38]: Ah, yes! On my mac this improved the time significantly. I'll take a note to compare it the Sanity Check time before and after.

**Comment by lidizheng** [comment]
CC @gnossen for the Python formatter change.

@sergiitk can't say I like the added restriction for the list expand style, but as long as the format is consistent, I'm happy.

**Comment by sergiitk** [comment]
@lidizheng agree, not too wild about it myself. But helping yapf to distinguish between parens (which can be ambiguous) and square brackets (clearly implying an array) fixed it.
As an alternative, I can move a the generator to a variable on the line above. I tried that, and it worked, but this didn't help readability either.
Yapf is weird.

**Review by gnossen — APPROVED** [review]

**Pull request merged**